test/controllers/users_controller_test.rb

```diff
@@ -1,4 +1,80 @@
 require "test_helper"
 
 describe UsersController do
+  let(:dan) { users(:dan) }
+  let(:kari) { users(:kari) }
+
+  describe 'auth_callback' do
+    it "should log in an existing user" do
+      start_count = User.count
+      perform_login(dan)
+
+      must_redirect_to root_path
+      session[:uid].must_equal dan.uid
+      expect(start_count).must_equal User.count
+    end
+      
+    it "should create a new user" do
+      start_count = User.count
+      kari.uid = 2420
```

Comment:
do you need to re-assign Kari's uid if Kari's fixture already contains their uid? It seems you might not have saved this adjustment so that User uid was not changed.